components/uid.rst

```diff
@@ -0,0 +1,200 @@
+.. index::
+   single: UID
+   single: Components; UID
+
+The UID Component
+=================
+
+    The UID component provides utilities to work with `unique identifiers`_ (UIDs)
+    such as UUIDs and ULIDs.
+
+.. versionadded:: 5.1
+
+    The UID component was introduced in Symfony 5.1 as an
+    :doc:`experimental feature </contributing/code/experimental>`.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/uid
+
+.. include:: /components/require_autoload.rst.inc
+
+UUIDs
+-----
+
+`UUIDs`_ (*universally unique identifiers*) are one of the most popular UIDs in
+the software industry. UUIDs are 128-bit numbers usually represented as five
+groups of hexadecimal characters: ``xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx``
+(the ``M`` digit is the UUID version and the ``N`` digit is the UUID variant).
+
+Generating UUIDs
+~~~~~~~~~~~~~~~~
+
+Use the named constructors of the ``Uuid`` class or any of the specific classes
+to create each type of UUID::
+
+    use Symfony\Component\Uid\Uuid;
+    use Symfony\Component\Uid\UuidV1;
+    use Symfony\Component\Uid\UuidV3;
+    use Symfony\Component\Uid\UuidV4;
+    use Symfony\Component\Uid\UuidV5;
+    use Symfony\Component\Uid\UuidV6;
+
+    // UUID type 1 generates the UUID using the MAC address of your device and a timestamp.
+    // Both are obtained automatically, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v1();
+    $uuid = new UuidV1();
+
+    // UUID type 4 generates a random UUID, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v4();
```

Comment:
here is 4

Reply:
I think UUIDv1 and UUIDv4 (especially v4) are the most commonly used, so I think it's best to have them before v2 and v3.